### PR TITLE
Build refactor

### PR DESCRIPTION
**Pull request by jroper**
* Switched to sbt-doge cross building
* Removed use of system properties
* Cleaned up build somewhat
* Made documentation build use cross building
* Made templates build use cross building
* Remove system properties from templates build
* Added templates command for publishing a specific template
* Added custom release process
* Added quick publish command that disables building docs and source jars

### CODE REVIEW

**Comment by jroper** [comment]
:( @play-pull-request-validator maybe no timeouts this time?

**Comment by benmccann** [comment]
i always like seeing 200 lines of code simplified out :-)  fyi, there's a merge conflict

**Comment by jroper** [comment]
Build passes. On the negative side, the build now takes 45 minutes, because we're now cross building the tests, documentation tests and templates.  We could reduce that time by not cross building the templates or the documentation tests, any opinions?

File: framework/project/Publish.scala [hunk 1]
```diff
@@ -65,35 +61,19 @@ object PublishSettings {
...
-          } else {
-            ResolverSettings.publishTypesafeIvyReleases
-          })
-        },
-        publishMavenStyle := false
-      )
-    }
+    commonPublishSettings ++ Seq(
+      publishTo := {
+        Some(if (isSnapshot(version.value)) {
```
Comment (pvlugter) [inline framework/project/Publish.scala]: There's also an `isSnapshot` setting built into sbt.

Comment (jroper) [inline framework/project/Publish.scala]: Updated.

**Comment by pvlugter** [comment]
Cool. Good stuff.

It could be nicer to have a faster validation build. Should we just build against 2.11 since that's the default in the templates? We've also got the master branch build, which could do the full cross build.

**Comment by jroper** [comment]
The second build actually only took 36 minutes.  Anyway, I've updated it so it now only builds against Scala 2.11.5.

**Comment by jroper** [comment]
Ok, now we're back to 23 minutes.

**Comment by yanns** [comment]
:+1:

**Pull request merged**